service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOControllerTest.java

```diff
@@ -0,0 +1,207 @@
+            convertdateTopLocalFormat(date.plusDays(5)));
+
+        actAndAssert(request, expected);
+    }
+
+    private void actAndAssert(CallbackRequest request, List<String> expected) throws Exception {
+        MvcResult response = makeRequest(request, "about-to-start");
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = mapper.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        assertThat(getReturnedDatesFromResponse(callbackResponse)).isEqualTo(expected);
+    }
+
+    @Test
+    void aboutToSubmitShouldPopulateHiddenHearingDateField() throws Exception {
+        List<Element<HearingBooking>> hearingDetails = createHearingBookings(date);
+
+        DynamicList dynamicHearingDates = draftCMOService.buildDynamicListFromHearingDetails(hearingDetails);
+
+        dynamicHearingDates
+            .setValue(
+                DynamicListElement.builder()
+                    .code(date.plusDays(5).toString())
+                    .label(date.plusDays(5).toString())
+                    .build());
+
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .data(ImmutableMap.of("cmoHearingDateList", dynamicHearingDates))
+                .build())
+            .build();
+
+        MvcResult response = makeRequest(request, "about-to-submit");
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = mapper.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        assertThat(caseData.getCaseManagementOrder().getHearingDate())
+            .isEqualTo(date.plusDays(5).toString());
+    }
+
+    private List<String> getReturnedDatesFromResponse(AboutToStartOrSubmitCallbackResponse callbackResponse) {
+        Map<String, Object> cmoHearingResponse = mapper.convertValue(
+            callbackResponse.getData().get("cmoHearingDateList"), Map.class);
+
+        List<Map<String, Object>> listItemMap = mapper.convertValue(cmoHearingResponse.get("list_items"), List.class);
+
+        return listItemMap.stream()
+            .map(element -> mapper.convertValue(element, DynamicListElement.class))
+            .map(DynamicListElement::getCode).collect(Collectors.toList());
+    }
+
+    private MvcResult makeRequest(CallbackRequest request, String endpoint) throws Exception {
+        return mockMvc
+            .perform(post("/callback/draft-cmo/" + endpoint)
+                .header("authorization", AUTH_TOKEN)
+                .header("user-id", USER_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(mapper.writeValueAsString(request)))
+            .andExpect(status().isOk())
+            .andReturn();
+    }
+
+    private String convertdateTopLocalFormat(LocalDate date) {
```

Comment:
could we just have this to be `convertDate`?

Reply:
Done this change.